Documentation/network/servicemesh/ingress-reference.rst

```diff
@@ -0,0 +1,160 @@
+these cases, when traffic arrives at the Service's port, eBPF code intercepts
+the traffic and transparently forwards it to Envoy (using the TPROXY kernel facility).
+
+This affects things like client IP visibility, which works differently for Cilium's
+Ingress and Gateway API support to other Ingress controllers.
+
+It also allows Cilium's Network Policy engine to apply CiliumNetworkPolicy to
+traffic bound for and traffic coming from an Ingress.
+
+Cilium's ingress config and CiliumNetworkPolicy
+***********************************************
+
+Ingress and Gateway API traffic bound to backend services via Cilium passes through a
+per-node Envoy proxy.
+
+The per-node Envoy proxy has special code that allows it to interact with the
+eBPF policy engine, and do policy lookups on traffic. This allows Envoy to be
+a Network Policy enforcement point, both for Ingress (and Gateway API) traffic,
+and also for east-west traffic via GAMMA or L7 Traffic Management.
+
+However, for ingress config, there's also an additional step. Traffic that arrives at
+Envoy *for Ingress or Gateway API* is assigned the special ``ingress`` identity
+in Cilium's Policy engine.
+
+Traffic coming from outside the cluster is usually assigned the ``world`` identity
+(unless there are IP CIDR policies in the cluster). This means that there are
+actually *two* logical Policy enforcement points in Cilium Ingress - before traffic
+arrives at the ``ingress`` identity, and after, when it is about to exit the
+per-node Envoy.
+
+.. image:: /images/ingress-policy.png
+    :align: center
+
+This means that, when applying Network Policy to a cluster, it's important to
+ensure that both steps are allowed, and that traffic is allowed from ``world`` to
+``ingress``, and from ``ingress`` to identities in the cluster (like the
+``productpage`` identity in the image above).
+
+Please see the Ingress and Policy example below for more details.
+
+Source IP Visibility
+********************
+
+.. Note::
+
+    By default, source IP visibility for Cilium ingress config, both Ingress
+    and Gateway API, should *just work* on most installations. Read this section
+    for more information on requirements and relevant settings.
+
+Having a backend be able to deduce what IP address the actual request came from
+is important for most applications.
+
+By default, Cilium's Envoy instances are configured to append the visible source
+address of incoming HTTP connections to the ``X-Forwarded-For`` header, using the
+usual rules. That is, by default Cilium sets the number of trusted hops to ``0``,
+indicating that Envoy should use the address the connection is opened from, rather
+than a value inside the ``X-Forwarded-For`` list. Increasing this count will
+have Envoy use the ``n`` th value from the list, counting from the right.
+
+Envoy will also set the ``X-Envoy-External-Address`` header to the trusted client
+address, whatever that turns out to be, based on ``X-Forwarded-For``.
+
+.. Note::
```

Comment:
I see. It makes total sense to highlight key ideas and tbh I don't see a better way of doing it than a note. It would be nice to have a different kind of a callout for highlights, but AFAIK we have only notes and warnings.